hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -427,23 +427,25 @@ object PruneDeadFields {
           })
         )
         memoizeMatrixIR(child, explodedDep, memo)
-      case MatrixAggregateRowsByKey(child, expr) =>
-        val irDep = memoizeAndGetDep(expr, requestedType.entryType, child.typ, memo)
-        val childDep = child.typ.copy(
-          globalType = unify(child.typ.globalType, irDep.globalType, requestedType.globalType),
-          rvRowType = irDep.rvRowType,
-          colType = unify(child.typ.colType, irDep.colType, requestedType.colType)
+      case MatrixAggregateRowsByKey(child, entryExpr, rowExpr) =>
+        val irDepEntry = memoizeAndGetDep(entryExpr, requestedType.entryType, child.typ, memo)
+        val irDepRow = memoizeAndGetDep(rowExpr, requestedType.rowValueStruct, child.typ, memo)
+        val childDep = unify(child.typ,
+          irDepEntry,
+          irDepRow,
+          minimal(child.typ).copy(globalType = requestedType.globalType, colType = requestedType.colType)
```

Comment:
looks good.

Reply:
You wrote it ;)

Reply:
ha, well I did spend 5 minutes making sure my past self was correct! :)

It's very hard to get these prune rules right 😞